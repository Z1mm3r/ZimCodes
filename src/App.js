import {createMuiTheme, MuiThemeProvider} from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import React, {useState, useEffect} from 'react'
import { Route, Switch, useLocation } from 'react-router-dom'
import {useTransition, animated} from '@react-spring/web'

import './App.css';
import AboutPage from './screens/AboutMePage'
import AboutSitePage from './screens/AboutSitePage'
import {DARK, LIGHT} from 'constants/colors'
import {darkTheme, lightTheme} from './themes'
import Header from './components/Header'
import IndexPage from './screens/IndexPage'
import LightingContext from './components/LightingContext'
import  {LOCATIONS } from 'constants/constants.js'
import {PAGE_LEFT_ENTER, PAGE_RIGHT_ENTER} from 'constants/transitions'
import ResumePage from './screens/ResumePage'

function App() {

  const [lighting,setLighting] = useState(LIGHT)

  //TODO debate pros and cons of this 
  //If we want React.strict mode... need to createMuiTheme everytime we want to swap from
  //light to dark.
  //IF we REMOVE React.strict... we can simply pass our themes in from the themes file
  //without using a themeState.
  //
  const [theme, setTheme] = useState(createMuiTheme(lightTheme))
  const toggleLighting = () => {
    setTheme( createMuiTheme (lighting === LIGHT ? darkTheme: lightTheme) )
    setLighting( lighting === LIGHT ? DARK : LIGHT)
  }
//----------------------Page Transition Logic--------------------------------------------------------------
  const location = useLocation();
  const [previousLocation, setPreviousLocation] = useState(location.pathname)
  const [entranceSide, setEntranceSide] = useState(PAGE_LEFT_ENTER)
  const [left,setLeft] = useState(true)
  //Using superSet to stop race conditions on our transition creations
  const [superSet, setSuperSet] = useState({
    location:location,
    previousLocation:previousLocation,
    left: left,
    entranceSide: entranceSide
   })

  useEffect(() => {
    if(LOCATIONS.indexOf(previousLocation) > LOCATIONS.indexOf(location.pathname)){
      setEntranceSide(PAGE_LEFT_ENTER)
      setLeft(true)
      setPreviousLocation(location.pathname)
    }
    else if(LOCATIONS.indexOf(previousLocation) < LOCATIONS.indexOf(location.pathname)){
      setEntranceSide(PAGE_RIGHT_ENTER)
      setLeft(false)
      setPreviousLocation(location.pathname)
    }
  }, [location])

  useEffect(() =>{
    setSuperSet({location,previousLocation,left,entranceSide})
  },[previousLocation])

  const transitions = useTransition(superSet.location,superSet.entranceSide,[superSet.location,superSet.entranceSide])[0]
//---------------------------------------------------------------------------------------

  return (
    <div className="App">
      <LightingContext.Provider value={{lighting,toggleLighting}}>
        <MuiThemeProvider theme={theme}>
            <CssBaseline />
            <Header/>
            {transitions((props,item) =>(
              <animated.div style={ props }>
                <Switch location={item}>
                  <Route exact path="/" component={IndexPage} />
                  <Route exact path="/about" component={AboutPage} />
                  <Route exact path="/resume" component={ResumePage} />
                  <Route exact path="/site" component={AboutSitePage} />
                </Switch>
              </animated.div>
            ))}
        </MuiThemeProvider  >
      </LightingContext.Provider>
    </div>
  );
}

export default App;
