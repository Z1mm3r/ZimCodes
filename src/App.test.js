import React from 'react'
import {render, screen} from '@testing-library/react'
import App from './App.js' 
import { MemoryRouter } from  'react-router-dom'

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000"
  }),
  useRouteMatch: () => ({ url: '/'}),
}));


test('renders without crashing', () => {
  const {asFragment} = render(<App />, {wrapper: MemoryRouter})
  expect(asFragment()).toMatchSnapshot();
})