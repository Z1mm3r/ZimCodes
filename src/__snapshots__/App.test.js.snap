// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="makeStyles-lightMode-2"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <a
                data-testid="navButtonLink"
                href="/"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Home
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <a
                data-testid="navButtonLink"
                href="/about"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    About
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <div
                class="border"
              >
                <div>
                  <div>
                    <img
                      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                      src="/Images/WIP-LOGO.png"
                      title="Zim Codes"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <a
                data-testid="navButtonLink"
                href="/resume"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Resume
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <a
                data-testid="navButtonLink"
                href="/site"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    About Site
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-lightModeLightbulb-3 MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3.55 18.54l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8zM11 22.45h2V19.5h-2v2.95zM4 10.5H1v2h3v-2zm11-4.19V1.5H9v4.81C7.21 7.35 6 9.28 6 11.5c0 3.31 2.69 6 6 6s6-2.69 6-6c0-2.22-1.21-4.15-3-5.19zm5 4.19v2h3v-2h-3zm-2.76 7.66l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      style="opacity: 1; transform: translate(0%,0);"
    />
  </div>
</DocumentFragment>
`;
